scripts/create-listings.js

```diff
@@ -0,0 +1,75 @@
+// This dotenv import is required for the `.env` file to be read
+require('dotenv').config();
+
+const flexIntegrationSdk = require('sharetribe-flex-integration-sdk');
+
+const integrationSdk = flexIntegrationSdk.createInstance({
+  // These two env vars need to be set in the `.env` file.
+  clientId: process.env.FLEX_INTEGRATION_CLIENT_ID,
+  clientSecret: process.env.FLEX_INTEGRATION_CLIENT_SECRET,
+
+  // Normally you can just skip setting the base URL and just use the
+  // default that the `createInstance` uses. We explicitly set it here
+  // for local testing and development.
+  baseUrl: process.env.FLEX_INTEGRATION_BASE_URL || 'https://flex-integ-api.sharetribe.com',
+});
+
+// Create mock listings for demo purposes. Replace this listings array
+// with real listings from e.g. an external integration. Make sure the 
+// user ids refer to existing users within Flex.
+const listings = [];
+const title = 'Listing #';
+const description = 'Listing created with Integration API';
+const userId = 'add-valid-user-id-here';
+const max = 105;
+
+for (let i = 0; i < max; i++) {
+  const count = i + 1;
+  listings.push({
+      title: `${title}${count}`,
+      description: `${description}`,
+      authorId: userId,
+      state: 'published',
+        price: { amount: 10000, currency: "USD" },
+      metadata: {
+        extId: count
+      }
+    });
+}
+
+const timeout = 500;
+
+const createWithTimeouts = (fns, resolve, reject, results = []) => {
```

Comment:
In an earlier version of the code you had a nice example of handling the rate limit and a wait time in case of 429. I think it's a nice idea to include it, even if just to highlight that implementations should be able to deal with that.

In the current version, the 500 ms timeout would mean that in ideal conditions, there could be close to 120 listings created per minute, which is likely to trigger the rate limit. In that sense, too, it feels nice to show an example of that.

If you prefer not to add code to deal with 429s, let's at least add a comment about why the pause is there and highlight the possibility of the client hitting the rate limit. In that case, it might be better to set the timeout to 600 to ensure that the create listing will be called at most 100 times a minute.

Reply:
Thanks for pointing that out! I extended the script now a bit, adding a full 429 handling version and including a "in case of 429" path in the timeout version as well.